File: Leetcode/JS/Leetcode748-最短补全单词.js
var shortestCompletingWord = function(licensePlate, words) {
    let mp = new Array(26).fill(0);
    for(const ch of licensePlate){
       if(/^[a-zA-Z]+$/.test(ch)) {
           mp[ch.toLocaleLowerCase().charCodeAt(0) - 'a'.charCodeAt(0)]++;
       }
    }
    let idx = -1;
    for(let i = 0;i < words.length;i++){
        const c = Array(26).fill(0);
        for(let j = 0;j < words[i].length;j++){
            const ch = words[i][j];
            ++c[ch.charCodeAt(0) - 'a'.charCodeAt(0)];
        }
        let ok = true;
        for(let j = 0;j < 26;j++){
            if(c[j] < mp[j]) {
                ok = false;
                break;
            }
        }
        if(ok && (idx < 0 || words[i].length < words[idx].length)) {
            idx = i;
        } 
    }
    return words[idx];
};

File: Leetcode/PY/Leetcode748-最短补全单词.py
class Solution:
    def shortestCompletingWord(self, licensePlate: str, words: List[str]) -> str:
        mp = defaultdict(int)
        for ch in licensePlate:
            if ch == ' ' or ch.isdigit(): continue
            mp[ch.lower()] += 1
        res = ''
        def check(mp,st):
            for k,v in mp.items():
                if st[k] < v: return False
            return True
        for stri in words:
            st = defaultdict(int)
            for ch in stri:
                st[ch] += 1
            if check(mp,st):
                if len(res) == 0 or len(res) > len(stri):
                    res = stri
        return res 